internal/builtin/data_rss_test.go

```diff
@@ -0,0 +1,305 @@
+package builtin
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/hcldec"
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+	"github.com/mmcdole/gofeed"
+	"github.com/stretchr/testify/assert"
+	"github.com/zclconf/go-cty/cty"
+
+	"github.com/blackstork-io/fabric/plugin"
+)
+
+func Test_makeRSSDataSchema(t *testing.T) {
+	t.Parallel()
+	t.Run("basic", func(t *testing.T) {
+		t.Parallel()
+
+		assert := assert.New(t)
+		schema := makeRSSDataSource()
+		assert.NotNil(schema.Config)
+		assert.NotNil(schema.Args)
+		assert.NotNil(schema.DataFunc)
+	})
+
+	t.Run("config_parses", func(t *testing.T) {
+		t.Parallel()
+
+		assert := assert.New(t)
+		schema := makeRSSDataSource()
+		cfg, diags := hclsyntax.ParseConfig([]byte(`
+			basic_auth {
+				username = "example_username"
+				password = "example_password"
+			}
+		`), "", hcl.InitialPos)
+		assert.Empty(diags)
+
+		val, diags := hcldec.Decode(cfg.Body, schema.Config, nil)
+		assert.Empty(diags)
+		username := val.GetAttr("basic_auth").Index(cty.NumberIntVal(0)).GetAttr("username").AsString()
+
+		assert.Equal("example_username", username, val.GoString())
+	})
+	t.Run("config_errors_if_partial", func(t *testing.T) {
+		t.Parallel()
+
+		assert := assert.New(t)
+		schema := makeRSSDataSource()
+		cfg, diags := hclsyntax.ParseConfig([]byte(`
+			basic_auth {
+				username = "example_username"
+			}
+		`), "", hcl.InitialPos)
+		assert.Empty(diags)
+
+		_, diags = hcldec.Decode(cfg.Body, schema.Config, nil)
+		assert.True(diags.HasErrors())
+	})
+
+	t.Run("config_ok_empty", func(t *testing.T) {
+		t.Parallel()
+
+		assert := assert.New(t)
+		schema := makeRSSDataSource()
+		cfg, diags := hclsyntax.ParseConfig([]byte(`
+
+		`), "", hcl.InitialPos)
+		assert.Empty(diags)
+
+		val, diags := hcldec.Decode(cfg.Body, schema.Config, nil)
+		assert.Empty(diags)
+		auth := val.GetAttr("basic_auth")
+		assert.Equal(0, auth.LengthInt())
+	})
+}
+
+func makeTestRssServer() (baseAddr string, closer func()) {
+	close := func() {}
+	defer func() {
+		close()
+	}()
+	lis, err := net.Listen("tcp4", "127.0.0.1:0")
```

Comment:
There is `net/http/httptest` standard package for testing http.  It will create a new listener and give you url:
```go
srv := httptest.NewServer(mux)
_ = srv.URL
```